DataDog/chaos-controller
# Retry to cleanup if it fails

### What does this PR do?

It adds a retry with an exponential backoff to the injector cleanup method call.

### Motivation

Retrying cleanup when it fails is a best effort way to eventually succeed the disruption cleanup. While it probably won't solve every cases of a failing cleanup, it can do the trick for some edge cases.

### Testing Guidelines

The best way to try that is to add a random failure in the injector you want to try. You can also add a permanent failure to see the retry running 3 times and exiting with a non-zero code.
